fieldtofile/fieldtofile-service.ts

```diff
@@ -1,5 +1,21 @@
 import * as fs from "fs";
 import * as PH from "processhub-sdk";
+import http from "http";
+
+async function writeFile(downloadUrl: string, targetpath: string): Promise<void> {
+  const file = fs.createWriteStream(targetpath);
+
+  return await new Promise<void>((resolve, reject) => {
+    http.get(downloadUrl, function (response) {
+      if (response.statusCode === 200) {
+        response.pipe(file);
+        resolve();
+      } else {
+        reject(response.statusCode);
```

Comment:
`file` needs to be closed in this case I think, otherwise this will create a memory leak.

Reply:
`file.end()` is now called at the end of the function.

Reply:
I'm not sure what happens if `end()` is called even if pipe was successful. As far as I remember, you would have to only close it in the error case (because if the inbound stream ends the outbound stream is also closed automatically) but I could also be wrong. If you are sure thats a no-op in this case, it's fine.